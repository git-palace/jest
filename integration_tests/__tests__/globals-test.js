/**
 * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 */

'use strict';

const path = require('path');
const runJest = require('../runJest');
const skipOnWindows = require('skipOnWindows');

skipOnWindows.suite();

const dir = path.resolve(__dirname, '../globals');

test('global jest variables', () => {
  const {stderr, status} = runJest(dir);

  expect(status).toBe(0);

  const output = stderr
    .split('\n')
    .slice(0, -2)
    .join('\n')
    .replace(/\s*\(.*ms\)/gm, '');

  expect(output).toMatchSnapshot();
});
