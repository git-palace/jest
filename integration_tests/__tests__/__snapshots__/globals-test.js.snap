exports[`test basic test constructs 1`] = `
" PASS  __tests__/basic-test-constructs-test.js
  ✓ it
  ✓ test
  describe
    ✓ it
    ✓ test

"
`;

exports[`test basic test constructs 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       4 passed, 4 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`test only 1`] = `
" PASS  __tests__/only-constructs-test.js
  ✓ test.only
  ✓ it.only
  ✓ fit
  ○ skipped 1 test
  fdescribe
    ✓ it
    ✓ test
  describe.only
    ✓ test
    describe
      ✓ test

"
`;

exports[`test only 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       1 skipped, 7 passed, 8 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`test only with expand arg 1`] = `
" PASS  __tests__/only-constructs-test.js
  ✓ test.only
  ✓ it.only
  ✓ fit
  ○ it
  fdescribe
    ✓ it
    ✓ test
  describe.only
    ✓ test
    describe
      ✓ test

"
`;

exports[`test only with expand arg 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       1 skipped, 7 passed, 8 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`test skips 1`] = `
" PASS  __tests__/skips-constructs-test.js
  ✓ it
  ○ skipped 4 tests
  xdescribe
    ○ skipped 2 tests
  describe.skip
    ○ skipped 1 test
    describe
      ○ skipped 1 test

"
`;

exports[`test skips 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       8 skipped, 1 passed, 9 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`test skips with expand arg 1`] = `
" PASS  __tests__/skips-constructs-test.js
  ✓ it
  ○ xtest
  ○ xit
  ○ it.skip
  ○ test.skip
  xdescribe
    ○ it
    ○ test
  describe.skip
    ○ test
    describe
      ○ test

"
`;

exports[`test skips with expand arg 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       8 skipped, 1 passed, 9 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`test tests with no implementation 1`] = `
" PASS  __tests__/only-constructs-test.js
  ✓ it
  ○ skipped 2 tests

"
`;

exports[`test tests with no implementation 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       2 skipped, 1 passed, 3 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;

exports[`test tests with no implementation with expand arg 1`] = `
" PASS  __tests__/only-constructs-test.js
  ✓ it
  ○ it, no implementation
  ○ test, no implementation

"
`;

exports[`test tests with no implementation with expand arg 2`] = `
"Test Suites: 1 passed, 1 total
Tests:       2 skipped, 1 passed, 3 total
Snapshots:   0 total
Time:        <<REPLACED>>
Ran all test suites.
"
`;
